bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java

```diff
@@ -123,7 +122,7 @@ public GroupFunction getFunction() {
      * @return all members of this and all contained {@link GroupItem}s
      */
     public Set<Item> getAllMembers() {
-        return ImmutableSet.copyOf(getMembers((Item i) -> !(i instanceof GroupItem)));
+        return Collections.unmodifiableSet(getMembers((Item i) -> !(i instanceof GroupItem)));

```

Comment:
to respect the former `copyOf`, should this also be wrapped in `new HashSet(...)`?

Reply:
Done, and to also make travis happy I used `LinkedHashSet` to preserve the item order.